core/thread.py

```diff
@@ -408,64 +463,65 @@ def __getitem__(self, item):
         searching channel history for genesis embed and
         extracts user_id from that.
         """
-        user_id = None
+        user_id = -1
 
-        if channel.topic and 'User ID: ' in channel.topic:
-            user_id = int(re.findall(r'\d+', channel.topic)[0])
+        if channel.topic:
+            user_id = match_user_id(channel.topic)
 
         # BUG: When discord fails to create channel topic.
         # search through message history
         elif channel.topic is None:
-            async for message in channel.history(limit=50):
+            async for message in channel.history(limit=100):
                 if message.embeds:
-                    em = message.embeds[0]
-                    # TODO: use re.search instead
-                    matches = re.findall(r'User ID: (\d+)', str(em.footer.text))
-                    if matches:
-                        user_id = int(matches[0])
-                        break
-
-        if user_id is not None:
+                    embed = message.embeds[0]
+                    if embed.footer.text:
+                        user_id = match_user_id(embed.footer.text)
+                        if user_id != -1:
+                            break
+
+        if user_id != -1:
             if user_id in self.cache:
                 return self.cache[user_id]
 
             recipient = self.bot.get_user(user_id)  # this could be None
+            if recipient is None:
```

Comment:
Shouldn't raise an error if a recipient is not found. Things like closing a thread and taking notes should still work if the recipient no longer is there.

Reply:
Yep, resolved.